logsmanagement/db_api.c

```diff
@@ -0,0 +1,1096 @@
+/** @file db_api.c
+ *  @brief This is the file containing the database API
+ *
+ *  @author Dimitris Pantazis
+ */
+
+#include "../daemon/common.h"
+#include "db_api.h"
+#include <inttypes.h>
+#include <stdio.h>
+#include "circular_buffer.h"
+#include "compression.h"
+#include "helper.h"
+#include "lz4.h"
+#include "parser.h"
+
+#define MAIN_DB "main.db" /**< Primary DB with just 1 table - MAIN_COLLECTIONS_TABLE **/
+#define MAIN_COLLECTIONS_TABLE "LogCollections"
+#define BLOB_STORE_FILENAME "logs.bin"
+#define METADATA_DB_FILENAME "metadata.db"
+#define LOGS_TABLE "Logs"
+#define BLOBS_TABLE "Blobs"
+
+static uv_loop_t *db_loop;
+static sqlite3 *main_db;
+static char *main_db_dir;  /**< Directory where all the log management databases and log blobs are stored in **/
+static char *main_db_path; /**< Path of MAIN_DB **/
+
+void db_set_lock(uv_mutex_t *db_mut) { 
+	uv_mutex_lock(db_mut); 
```

Comment:
Tabulation in this file looks like a tab instead 4 spaces.

Reply:
Considering that the two initial functions are calls to an unique function in `libuv` library,  I suggest either remove them or to convert them to `inline` functions.

Reply:
Done!